### PR TITLE
Replace language sidebar with dedicated language screen

### PR DESCRIPTION
**Pull request by xmrdsc**
Due to a hard to debug buildbot issue concerning the sidebar on a statically compiled Qt 5.7.0, I concluded the pragmatic solution is to temporarily disable the sidebar for now until we migrate to > Qt 5.9.

### CODE REVIEW

File: wizard/WizardLang.qml [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        opacity: 1.0
+        anchors.fill: parent
+        source: "../images/middlePanelBg.jpg"
+    }
+
+    ColumnLayout {
+        anchors.top: parent.top
+        anchors.topMargin: persistentSettings.customDecorations ? 90 * scaleRatio : 32 * scaleRatio
+        Layout.alignment: Qt.AlignHCenter;
+        width: parent.width - 100
```
Comment (xiphon) [inline wizard/WizardLang.qml]: `scaleRatio`

File: wizard/WizardLang.qml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            return 0;
+        } else if(appWindow.width < 800){
+            return 1;
+        } else {
+            return 2;
+        }
+    }
+
+    Image {
+        opacity: 1.0
```
Comment (xiphon) [inline wizard/WizardLang.qml]:
```suggestion
```

File: wizard/WizardLang.qml [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+            readOnly: true
+        }
+
+        Flow {
+            id: flow
+            height: 800 * scaleRatio
+            Layout.fillWidth: true
+            Layout.topMargin: 20 * scaleRatio
+
+            spacing: 5
```
Comment (xiphon) [inline wizard/WizardLang.qml]: scaleRatio

File: wizard/WizardLang.qml [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+                        anchors.top: parent.top
+                        color: "transparent"
+                        width: 32 * scaleRatio
+                        height: parent.height
+
+                        Image {
+                            source: flag
+                            mipmap: true
+                            smooth: true
+                            width: 32
```
Comment (xiphon) [inline wizard/WizardLang.qml]: scaleRatio

File: wizard/WizardLang.qml [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+                        }
+                    }
+
+                    Rectangle {
+                        id: langRect
+                        anchors.top: parent.top
+                        anchors.left: img.right
+                        color: "transparent"
+                        height: parent.height
+                        width: langText.width + 22
```
Comment (xiphon) [inline wizard/WizardLang.qml]: scaleRatio

File: wizard/WizardLang.qml [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+        anchors.fill: parent
+        source: "../images/middlePanelBg.jpg"
+    }
+
+    ColumnLayout {
+        anchors.top: parent.top
+        anchors.topMargin: persistentSettings.customDecorations ? 90 * scaleRatio : 32 * scaleRatio
+        Layout.alignment: Qt.AlignHCenter;
+        width: parent.width - 100
+        Layout.fillWidth: true
```
Comment (xiphon) [inline wizard/WizardLang.qml]:
```suggestion
```

File: wizard/WizardLang.qml [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+    Image {
+        opacity: 1.0
+        anchors.fill: parent
+        source: "../images/middlePanelBg.jpg"
+    }
+
+    ColumnLayout {
+        anchors.top: parent.top
+        anchors.topMargin: persistentSettings.customDecorations ? 90 * scaleRatio : 32 * scaleRatio
+        Layout.alignment: Qt.AlignHCenter;
```
Comment (xiphon) [inline wizard/WizardLang.qml]:
```suggestion
        anchors.horizontalCenter: parent.horizontalCenter
```

File: wizard/WizardLang.qml [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+            textMargin: 0
+            leftPadding: 0
+            topPadding: 0
+            bottomPadding: 0
+            readOnly: true
+        }
+
+        TextArea {
+            Layout.fillWidth: true
+            anchors.horizontalCenter: parent.horizontalCenter
```
Comment (xiphon) [inline wizard/WizardLang.qml]:
```suggestion
            Layout.alignment: Qt.AlignHCenter;
```

**Comment by xiphon** [comment]
Would be better if the flags were aligned. Might be worth it to set `Repeater -> delegate: Rectangle -> width` to some fixes value

**Comment by erciccione** [comment]
> Would be better if the flags were aligned

I agree, but if it's just temporary for this release i don't mind much

File: wizard/WizardLang.qml [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+                            // reload active translations
+                            console.log(locale_spl[0]);
+                            translationManager.setLanguage(locale_spl[0]);
+
+                            // set wizard language settings
+                            wizard.language_locale = locale;
+                            wizard.language_wallet = wallet_language;
+                            wizard.language_language = display_name + " (" + locale_spl[1] + ") ";
+
+                            appWindow.showStatusMessage(qsTr("Language changed."), 3);
```
Comment (mmbyday) [inline wizard/WizardLang.qml]:
```suggestion
                            appWindow.showStatusMessage(qsTr("Language changed."), 3);
                            appWindow.toggleLanguageView();
```

Better user experience.

File: wizard/WizardHome.qml [hunk 10]
```diff
@@ -153,9 +153,7 @@ Rectangle {
                     text: qsTr("Change language") + translationManager.emptyString
 
                     onClicked: {
-                        wizardController.skipModeSelection = true;
-                        wizardController.wizardState = 'wizardLanguage';
-                        languageSidebar.open();
+                        appWindow.toggleLanguageView();
```
Comment (mmbyday) [inline wizard/WizardHome.qml]: The settings > layout > change language button needs updating too.

Comment (xmrdsc) [reply wizard/WizardHome.qml]: Thanks, looks like I did not commit this file.

File: wizard/WizardLang.qml (hunk 9 above)
Comment (xmrdsc) [reply wizard/WizardLang.qml]: Good catch.

File: wizard/WizardHome.qml (hunk 10 above)
Comment (xmrdsc) [reply wizard/WizardHome.qml]: Thanks

**Comment by xmrdsc** [comment]
> Would be better if the flags were aligned. Might be worth it to set `Repeater -> delegate: Rectangle -> width` to some fixes value



> Scaleratios

Fixed

> The settings > layout > change language button needs updating too.

Fixed

**Pull request merged**